ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/video/VideoView.kt

```diff
@@ -0,0 +1,528 @@
+            if (!released) throw e
+        }
+    }
+
+    private fun applySizing() {
+        @Suppress("ComplexCondition")
+        if (released || width == 0 || height == 0 || videoWidth == 0 || videoHeight == 0) return
+
+        val parentW = width
+        val parentH = height
+        val parentAR = parentW.toFloat() / parentH
+        val videoAR = videoWidth.toFloat() / videoHeight
+
+        val lp = texture.layoutParams as LayoutParams
+        if (scaleType == ScaleType.FIT) {
+            if (videoAR > parentAR) {
+                lp.width = parentW
+                lp.height = (parentW / videoAR).toInt()
+            } else {
+                lp.height = parentH
+                lp.width = (parentH * videoAR).toInt()
+            }
+        } else {
+            if (videoAR > parentAR) {
+                lp.height = parentH
+                lp.width = (parentH * videoAR).toInt()
+            } else {
+                lp.width = parentW
+                lp.height = (parentW / videoAR).toInt()
+            }
+        }
+        lp.gravity = Gravity.CENTER
+        texture.layoutParams = lp
+        texture.visibility = VISIBLE
+        texture.alpha = 1f
+        invalidate()
+    }
+
+    private fun safeSeekTo(position: Int) {
+        safely(execute = {
+            if (prepared && position > 0) player.seekTo(position)
+        })
+    }
+
+    /** MediaController.MediaPlayerControl **/
+    override fun start() {
+        if (prepared && !released) {
+            safely(execute = {
+                player.start()
+            })
+        }
+    }
+
+    override fun pause() {
+        if (prepared && !released) {
+            safely(execute = {
+                player.pause()
+            })
+        }
+    }
+
+    override fun getDuration(): Int = if (prepared && !released) player.duration else 0
+    override fun getCurrentPosition(): Int =
+        if (prepared && !released) player.currentPosition else resumePosMs
+
+    override fun seekTo(pos: Int) {
+        resumePosMs = pos
+        safeSeekTo(pos)
+    }
+
+    override fun isPlaying(): Boolean = prepared && !released && player.isPlaying
+    override fun getBufferPercentage(): Int = 0
+    override fun canPause(): Boolean = true
+    override fun canSeekBackward(): Boolean = true
+    override fun canSeekForward(): Boolean = true
+    override fun getAudioSessionId(): Int = player.audioSessionId
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        release()
+    }
+
+    @Suppress("ComplexCondition")
+    fun startIfNeeded() {
+        if (prepared && !released && !player.isPlaying && autoStart) {
+            start()
+        }
+    }
+}
+
+@Stable
+private data class SavedPlayback(val position: Int, val play: Boolean)
+
+// Simple static state holder as fallback when rememberSaveable fails
+private object VideoStateHolder {
+    private val states = mutableMapOf<String, SavedPlayback>()
```

Comment:
Do we need to clear the saved state here? I think right now if leaving the paywall and coming back, it would remain on the previous state right? In that scenario I think it should start from the beginning... 

I'm also wondering if we could save this state in a ViewModel instead of static like here (view models survive reconfiguring the activity)... But I think we can try to improve this in follow-up PRs.